instrumentation-core/spring/spring-webmvc-3.1-core/src/main/java/io/opentelemetry/instrumentation/springwebmvc/SpringWebMvcDecorator.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.springwebmvc;
+
+import io.grpc.Context;
+import io.opentelemetry.auto.bootstrap.instrumentation.decorator.HttpServerTracerBase;
+import io.opentelemetry.auto.instrumentation.servlet.HttpServletRequestGetter;
+import io.opentelemetry.context.propagation.HttpTextFormat.Getter;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Status;
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.servlet.http.HttpServletRequest;
+
+public class SpringWebMvcDecorator
+    extends HttpServerTracerBase<HttpServletRequest, HttpServletRequest, HttpServletRequest> {
+
+  public static final SpringWebMvcDecorator DECORATE = new SpringWebMvcDecorator();
+
+  public Span afterStart(final Span span) {
+    assert span != null;
+    return span;
+  }
+
+  public Span beforeFinish(final Span span) {
+    assert span != null;
+    return span;
+  }
+
+  @Override
+  public void onError(final Span span, final Throwable throwable) {
```

Comment:
There is no need to override and expose this method. Clients should use `end` or `endExceptionally` from the base class instead.